src/assets/javascripts/angular/services/academicsService.js

```diff
@@ -183,6 +183,43 @@ angular.module('calcentral.services').service('academicsService', function() {
     return classes;
   };
 
+  var getSwapClasses = function(enrolledCourses, waitlistedCourses) {
+    var count=0;
+    var swapClassList = [];
+    // initialize swap_number
+    for (var i=0; i<enrolledCourses.length; i++) {
+      enrolledCourses[i].swap_number = 0;
+    }
+    for (var i=0; i<waitlistedCourses.length; i++) {
```

Comment:
@fp2020 - On line 4 of this file we include the [LoDash library](https://lodash.com/) as the variable `_`. You can use this library to do various things like iterate over an array or objects properties. 

In this case you can use [forEach()](https://lodash.com/docs/4.17.14#forEach)

```javascript
_.forEach(enrolledCourses, function(course) {
  course.swap_number = 0;
});
_.forEach(waitlistedCourses, function(course) {
  course.swap_number = 0;
});
```

Reply:
We also use it often to get a value deep within a hierarchy, such as: `_.get(object, 'a[0].b.c');`

Reply:
Thanks! I will switch to more functional programming.